frontend/document/credocument.lua

```diff
@@ -365,14 +365,22 @@ function CreDocument:drawCurrentView(target, x, y, rect, pos)
     end
     -- TODO: self.buffer could be re-used when no page/layout/highlights
     -- change has been made, to avoid having crengine redraw the exact
-    -- same buffer.
-    -- And it could only change when some other methods from here are called
-    --local start_clock = os.clock()
-    self._drawn_images_count, self._drawn_images_surface_ratio = self._document:drawCurrentPage(self.buffer, self.render_color)
-    --print(string.format("CreDocument:drawCurrentView: Rendering took %9.3f ms", (os.clock() - start_clock) * 1000))
-    --start_clock = os.clock()
+    -- same buffer. And it could only change when some other methods
+    -- from here are called
+
+    -- If in night mode, ask crengine to invert all images, so they
+    -- get displayed in their original colors when the whole screen
+    -- is inverted by night mode
+    local invert_images = G_reader_settings:isTrue("night_mode")
+
+    -- local start_clock = os.clock()
+    self._drawn_images_count, self._drawn_images_surface_ratio =
+        self._document:drawCurrentPage(self.buffer, self.render_color, invert_images)
+    -- print(string.format("CreDocument:drawCurrentView: Rendering took %9.3f ms", (os.clock() - start_clock) * 1000))
+
+    -- start_clock = os.clock()
     target:blitFrom(self.buffer, x, y, 0, 0, rect.w, rect.h)
-    --print(string.format("CreDocument:drawCurrentView: Blitting took  %9.3f ms", (os.clock() - start_clock) * 1000))
+    -- print(string.format("CreDocument:drawCurrentView: Blitting took  %9.3f ms", (os.clock() - start_clock) * 1000))
```

Comment:
Whitespace :D.

(Extreme nitpicking aside, that was to differentiate between comments and commented out code ;)).

Reply:
Looking at the file and not the diff, I see those aren't right next to comments anymore, so, yeah, that works, too, never mind me ;).

Reply:
Interesting concept. I've spotted it but I thought it was just an inconsistency. ;-)

(I do in fact keep myself from posting nitpicks now and then. ;-D)

Reply:
It was extra hard to read to me with no whitespace :)
(I would have remove it if you just hadn't added it :) I think profiling code is just as easy to add and remove only when needed. Can/should I ?)

Reply:
That was mainly to be able to do it on-device over SSH via nano (instead of a patch/stash + scp from the desktop) ;).

Reply:
Or may be we could just have somewhere a function available for quick profiling, to avoid losing a few minutes to think about os.clock and string.format, that we could use as:
```diff
+ timeit("rendering document", function()
     original code to time
+ end)
```
or something even easier if some Lua constructs allow for easier, like C define...

Reply:
Yeah, I'm not terribly fond the the "wrap it in an anonymous func" approach usually taken for that kind of thing in Lua, but that's just me not being fond of many many Lua idiosyncracies in general ;).

Reply:
But yeah, having that kind of thing available in any manner somewhere would defintely come in handy ;).